src/optimizer/remove_unused_columns.cpp

```diff
@@ -287,6 +289,16 @@ void RemoveUnusedColumns::VisitOperator(LogicalOperator &op) {
 	}
 }
 
+static idx_t GetColumnIdsIndexForFilter(vector<ColumnIndex> &column_ids, idx_t filter_idx) {
```

Comment:
Factored this existing logic out into a method (see `RemoveColumnsFromLogicalGet`)